spec/unit/provider/nameservice_spec.rb

```diff
@@ -301,4 +301,42 @@
     end
   end
 
+  describe "comments_in_sync?(is,should)" do
+    it "should force_encoding of should-value to binary if is-value is binary" do
+      is_value = String.new("foo", :encoding => Encoding::ASCII_8BIT)
+      should_value = String.new("bar", :encoding => Encoding::UTF_8)
+
+      expect(should_value.encoding).to be(Encoding::UTF_8)
+      provider.comments_insync?(is_value, should_value)
+      expect(should_value.encoding).to be(Encoding::ASCII_8BIT)
+    end
+
+    it "should not modify encoding of should-value if is-value is not in binary" do
+      is_value = String.new("foo", :encoding => Encoding::ASCII)
+      should_value = String.new("bar", :encoding => Encoding::UTF_8)
+
+      expect(should_value.encoding).to be(Encoding::UTF_8)
+      provider.comments_insync?(is_value, should_value)
+      expect(should_value.encoding).to be(Encoding::UTF_8)
+    end
+
+    it "should do basic string comparison of the is-value with the should-value" do
+      is_value = "foo"
+      should_value = "bar"
+      is_value.expects(:'==').with("bar")
+      provider.comments_insync?(is_value, should_value)
+    end
+
+    it "should return false if the is-value and should-value are not equal" do
+      is_value = "foo"
+      should_value = "bar"
+      expect(provider.comments_insync?(is_value, should_value)).to be_falsey
+    end
+
+    it "should return true if only the is-value is in binary encoding but the strings are otherwise identical" do
+      is_value = String.new("foo", :encoding => Encoding::ASCII_8BIT)
```

Comment:
Apparently this form of the string constructor didn't exist until ruby 2.3.0 - updating